build_tools/python_deploy/build_windows_ci.sh

```diff
@@ -1,8 +1,30 @@
 #!/usr/bin/env bash
-set -eo pipefail
+set -eu -o errtrace
 
 echo "Building torch-mlir"
 
+cache_dir="${cache_dir:-}"
+this_dir="$(cd $(dirname $0) && pwd)"
+repo_root="$(cd ${this_dir}/../.. && pwd)"
+
+# Setup cache dir.
+if [ -z "${cache_dir}" ]; then
+  cache_dir="${repo_root}/.build-cache"
+  mkdir -p "${cache_dir}"
+  cache_dir="$(cd ${cache_dir} && pwd)"
+fi
+
+echo "Caching to ${cache_dir}"
+mkdir -p "${cache_dir}/ccache"
+mkdir -p "${cache_dir}/pip"
+
+export CCACHE_DIR="${cache_dir}/ccache"
+export CCACHE_MAXSIZE="350M"
```

Comment:
IDK if this will be enough cache space for the build... I guess we'll see after a few runs. Pretty low impact if there are cache misses given the run frequency we're starting that.